### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName is falsely reported on conflicting enums

### ISSUE THREAD

**Issue by mebigfatguy** [pmd/pmd#6271]
**Affects PMD Version:** 
7.18.0

**Rule:**
UnnecessaryFullyQualifiedName

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_XXX_XXX.html#XXX>

**Description:**

When two enum classes are present with overlapping enum values, one must be fully qualified

**Code Sample demonstrating the issue:**

```java
enum SoundsThatStartsWithF {

    Fee,
    Fi,
    Fo,
    Fum;

    public static GiantSounds convertSound(SoundsThatStartsWithF fSounds) {
        return switch (fSounds) {
            case Fee -> GiantSounds.Fee;
            case Fi -> GiantSounds.Fi;
            case Fo -> GiantSounds.Fo;
            case Fum -> GiantSounds.Fum;
        };
    }
}
```

**Expected outcome:**
PMD should not report this

PMD reports a violation at line "case Fee -> GiantSounds.Fee"  but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
ANT
